--- polkit-kde-agent-1-0.99.1/policykitlistener.cpp.dbusspam	2013-03-11 13:41:16.000000000 +0100
+++ polkit-kde-agent-1-0.99.1/policykitlistener.cpp	2013-10-25 15:43:44.561867136 +0200
@@ -26,6 +26,7 @@
 
 #include <PolkitQt1/Agent/Listener>
 #include <PolkitQt1/Agent/Session>
+#include <PolkitQt1/Authority>
 #include <PolkitQt1/Subject>
 #include <PolkitQt1/Identity>
 #include <PolkitQt1/Details>
@@ -57,7 +58,13 @@ PolicyKitListener::~PolicyKitListener()
 void PolicyKitListener::setWIdForAction(const QString& action, qulonglong wID)
 {
     kDebug() << "On to the handshake";
-    m_actionsToWID[action] = wID;
+    foreach(const PolkitQt1::ActionDescription &desc, PolkitQt1::Authority::instance()->enumerateActionsSync()) {
+        if (action == desc.actionId()) {
+            m_actionsToWID[action] = wID;
+            return;
+        }
+    }
+    kDebug() << "setWIdForAction:" << action << "was not found in the enumeration, not storing its WId";
 }
 
 void PolicyKitListener::initiateAuthentication(const QString &actionId,
